pandas/core/indexes/multi.py

```diff
@@ -277,16 +285,26 @@ def _verify_integrity(self, codes=None, levels=None):
                 raise ValueError("Unequal code lengths: %s" %
                                  ([len(code_) for code_ in codes]))
             if len(level_codes) and level_codes.max() >= len(level):
-                raise ValueError("On level %d, code max (%d) >= length of"
-                                 " level  (%d). NOTE: this index is in an"
-                                 " inconsistent state" % (i, level_codes.max(),
-                                                          len(level)))
+                raise ValueError("On level {level}, code max ({max_code})"
+                                 " >= length of level  ({level_len}). "
+                                 "NOTE: this index is in an inconsistent"
+                                 " state".format(
+                                     level=i, max_code=level_codes.max(),
+                                     level_len=len(level)))
+            if len(level_codes) and level_codes.min() < -1:
+                raise ValueError("On level {level}, code value ({code})"
+                                 " < -1".format(
+                                     level=i, code=level_codes.min()))
             if not level.is_unique:
                 raise ValueError("Level values must be unique: {values} on "
                                  "level {level}".format(
                                      values=[value for value in level],
                                      level=i))
 
+        codes = [self._validate_codes(level, code)
```

Comment:
can you return the codes here (and change the doc-string of _verify_integrity

Reply:
refactored corresponding code segments and doc-string updated